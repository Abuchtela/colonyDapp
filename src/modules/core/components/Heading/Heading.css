@import 'variables.css';

.main {
  composes: medium from '~styles/titles.css';
  margin-bottom: 20px;
  font-family: var(--family-primary);
  white-space: pre-line;
  color: var(--text);
}

/*
 * Theme-ing
 *
 * Next set of classes will define our various heading themes.
 */

/*
 * Primary is using the `colony` `green` color, as opposes as a `dark-grey` color for the heading.
 * This is done for consistency: everywhere we refer to something as `primary` we expect `colony` `green`.
 */
.primary {
  composes: main;
  color: var(--primary);
}

.dark {
  composes: main;
  color: var(--grey-1);
}

.invert {
  composes: main;
  color: var(--text-invert);
}

.uppercase {
  composes: main;
  text-transform: uppercase;
  color: var(--temp-grey-11);
}

/*
 * Margins
 *
 * Next set of classes will define the headings's margins
 */

.marginDouble {
  margin-bottom: 40px;
}

.marginNone {
  /* Need to overwrite the size class */
  margin-bottom: 0 !important;
}

/*
 * Sizing
 *
 * Next set of classes will define the headings's size
 */

.sizeHuge {
  composes: title from '~styles/titles.css';
  font-size: var(--size-huge);
}

.sizeLarge {
  composes: title from '~styles/titles.css';
  font-size: var(--size-large);
}

.sizeMediumL {
  composes: title from '~styles/titles.css';
  font-size: var(--size-medium-l);
}

.sizeNormal {
  composes: title from '~styles/titles.css';
  font-size: var(--size-normal);
}

.sizeSmall {
  composes: title from '~styles/titles.css';
  font-size: var(--size-small);
}

.sizeTiny {
  composes: title from '~styles/titles.css';
  font-size: var(--size-tiny);
  font-weight: var(--weight-thin);
}

.sizeThin {
  composes: thin from '~styles/titles.css';
}

.sizeBoldSmall {
  composes: boldSmall from '~styles/titles.css';
  margin-bottom: 4px;
  font-size: var(--size-small);
}

.sizeThinNormal {
  composes: thinNormal from '~styles/titles.css';
  font-size: var(--size-normal);
}

.sizeThinner {
  composes: thinner from '~styles/titles.css';
}
