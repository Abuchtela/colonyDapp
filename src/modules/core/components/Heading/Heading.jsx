/* @flow */

import React from 'react';
import type { Node } from 'react';

import { getMainClasses } from '~utils/css';

import type { IntlShape, MessageDescriptor } from 'react-intl';
import type { Appearance } from '~types/css';

import styles from './Heading.css';

const displayName = 'core.Heading';

type Props = {
  appearance?: Appearance,
  className?: string, // Overwriting class name(s). Setting this will overwrite appearance defaults.
  intl: IntlShape, // `react-intl` object, so that we have access to the `formatMessage()` method.
  tagName?: string, // String that will hard set the heading element to render (default: h3)
  children: Node, // Used to extend the functionality of the component. This will not generate a title attribute on the element.
  text: MessageDescriptor | string, // A string or a `MessageDescriptor` that make up the headings's text
  textValues?: { [string]: string }, // Values to interpolate in react-intl
};

// TODO: hoping this style will be fixed by prettier
const Heading = ({
  appearance,
  children,
  className,
  intl,
  tagName,
  text,
  textValues,
  ...props
}: Props) => {
  const size = (appearance && appearance.size) || 'medium';
  const HeadingElement =
    tagName ||
    {
      huge: 'h1',
      thin: 'h1',
      large: 'h2',
      mediumL: 'h6',
      medium: 'h3',
      small: 'h4',
      tiny: 'h5',
    }[size] ||
    'h3';
  const value =
    typeof text == 'string' ? text : intl.formatMessage(text, textValues);

  return (
    <HeadingElement
      title={value}
      className={className || getMainClasses(appearance, styles)}
      {...props}
    >
      {value || children}
    </HeadingElement>
  );
};

Heading.displayName = displayName;

export default Heading;
