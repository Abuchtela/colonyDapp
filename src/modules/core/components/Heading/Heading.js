/* @flow */

import compose from 'recompose/compose';
import { injectIntl } from 'react-intl';

import Heading from './Heading.jsx';

export default compose(injectIntl)(Heading);
