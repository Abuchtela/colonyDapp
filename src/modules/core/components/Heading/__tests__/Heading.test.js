/* eslint-env jest */

import React from 'react';
import toJson from 'enzyme-to-json';

import { shallowWithIntl, mountWithIntl } from '~utils/test';

import Heading from '../Heading.jsx';

describe('Heading component', () => {
  const mockTitle = 'Hello Rick';
  test('Renders initial component', () => {
    const wrapper = shallowWithIntl(<Heading text={mockTitle} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  test('Works with MessageDescriptors', () => {
    const wrapper = shallowWithIntl(
      <Heading text={{ id: 'rick', defaultMessage: mockTitle }} />,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  test('Shows the title value', () => {
    const wrapper = shallowWithIntl(<Heading text={mockTitle} />);
    expect(wrapper.find('h3').prop('title')).toEqual(mockTitle);
    expect(wrapper.children().text()).toEqual(mockTitle);
  });
  test('Has default heading element props', () => {
    const wrapper = mountWithIntl(<Heading text={mockTitle} />);
    expect(wrapper.find('h3')).toHaveLength(1);
  });
  test('Correctly generates heading elements', () => {
    /*
     * Huge
     */
    let wrapper = mountWithIntl(
      <Heading appearance={{ size: 'large' }} text={mockTitle} />,
    );
    expect(wrapper.find('h1')).toHaveLength(1);
    /*
     * Thin
     */
    wrapper = mountWithIntl(
      <Heading appearance={{ size: 'mediumL' }} text={mockTitle} />,
    );
    expect(wrapper.find('h2')).toHaveLength(1);
    /*
     * Large
     */
    wrapper = mountWithIntl(
      <Heading appearance={{ size: 'medium' }} text={mockTitle} />,
    );
    expect(wrapper.find('h3')).toHaveLength(1);
    /*
     * Medium
     */
    wrapper = mountWithIntl(
      <Heading appearance={{ size: 'normal' }} text={mockTitle} />,
    );
    expect(wrapper.find('h4')).toHaveLength(1);
    /*
     * Small
     */
    wrapper = mountWithIntl(
      <Heading appearance={{ size: 'small' }} text={mockTitle} />,
    );
    expect(wrapper.find('h5')).toHaveLength(1);
    /*
     * Tiny
     */
    wrapper = mountWithIntl(
      <Heading appearance={{ size: 'tiny' }} text={mockTitle} />,
    );
    expect(wrapper.find('h6')).toHaveLength(1);
  });
  test('Can overwrite the heading element', () => {
    const eliteElement = 'h1337';
    const wrapper = shallowWithIntl(
      <Heading tagName={eliteElement} text={mockTitle} />,
    );
    expect(wrapper.find(eliteElement)).toHaveLength(1);
  });
  test('Can overwrite the className', () => {
    const className = 'hardCodedStyles';
    const wrapper = shallowWithIntl(
      <Heading className={className} text={mockTitle} />,
    );
    expect(wrapper.props().className).toEqual(className);
  });
});
