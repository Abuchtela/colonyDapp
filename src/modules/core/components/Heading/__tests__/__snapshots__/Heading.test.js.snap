// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading component Renders initial component 1`] = `
<core.Heading
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {},
      "now": [Function],
      "textComponent": "span",
    }
  }
  text="Hello Rick"
/>
`;

exports[`Heading component Works with MessageDescriptors 1`] = `
<core.Heading
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {},
      "now": [Function],
      "textComponent": "span",
    }
  }
  text={
    Object {
      "defaultMessage": "Hello Rick",
      "id": "rick",
    }
  }
/>
`;
